src/LibraryManager/Providers/FileSystem/FileSystemProvider.cs

```diff
@@ -8,52 +8,40 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Web.LibraryManager.Contracts;
+using Microsoft.Web.LibraryManager.LibraryNaming;
 using Microsoft.Web.LibraryManager.Resources;
 
 namespace Microsoft.Web.LibraryManager.Providers.FileSystem
 {
     /// <summary>Internal use only</summary>
-    internal class FileSystemProvider : IProvider
+    internal sealed class FileSystemProvider : BaseProvider
     {
         /// <summary>Internal use only</summary>
         public FileSystemProvider(IHostInteraction hostInteraction)
+            :base(hostInteraction, null)
         {
-            HostInteraction = hostInteraction;
         }
 
         /// <summary>
         /// The unique identifier of the provider.
         /// </summary>
-        public string Id { get; } = "filesystem";
-
-        /// <summary>
-        /// The NuGet Package id for the package including the provider for use by MSBuild.
-        /// </summary>
-        /// <remarks>
-        /// If the provider doesn't have a NuGet package, then return <code>null</code>.
-        /// </remarks>
-        public string NuGetPackageId { get; } = "Microsoft.Web.LibraryManager.Build";
-
-        /// <summary>
-        /// An object specified by the host to interact with the file system etc.
-        /// </summary>
-        public IHostInteraction HostInteraction { get; }
+        public override string Id => "filesystem";
```

Comment:
Should we make these constants (in all providers)? Tests use those strings if nothing else. In this case, I noticed one test uses "fileSystem" instead of "filesystem", so I it seems it would fail on Linux/Mac?

Reply:
The bad test is a good catch, I'll fix that.

There's a reasonable argument against exposing the constants: if you exclusively rely on the constants in tests, you'll only be testing that the constant field is used, not that the value of that field is correct.  There is an equally reasonable counter proposal to this: add a single unit test asserting the value of the constant, then use the constant in all the other tests.

It looks like there's a lot of places in tests where we've just hardcoded the provider names out of convenience.  Fixing these to all be consistent doesn't seem to provide value worth the time to go fix up all the uses.  I'll leave this as a future exercise if we do that level of code cleanup; it doesn't seem to add any value in this PR.

(TL;DR: I agree, but not going to address it in this PR)